crates/subspace-farmer-components/src/segment_reconstruction.rs

```diff
@@ -42,8 +40,17 @@ pub async fn recover_missing_piece<PG: PieceGetter>(
     let semaphore = &semaphore;
     let acquired_pieces_counter = &acquired_pieces_counter;
 
-    let pieces = (starting_piece_index..)
+    // We prioritize source pieces over parity pieces here
```

Comment:
If we have source pieces we don't need to use erasure coding to recover source data, somewhat obvious.

Previously we didn't have interleaving so it worked naturally, with interleaving I just made it equivalent to what it was and left a comment since it might be surprising why it was even necessary for someone who is reading for the first time and might think it is useless. Comment clarifies that it is indeed intentional.